ohara-manager/client/cypress/support/e2eCommands.js

```diff
@@ -15,3 +15,188 @@
+      await bkApi.create(broker);
+      await objectApi.create(broker);
+      // broker api will make sure the service is starting, we can skip the checking
+      const bkRes = await bkApi.start(broker);
+      if (!bkRes.errors) result.broker = bkRes.data;
+
+      const worker = {
+        name: workspaceName,
+        group: workerGroup,
+        nodeNames: [node.hostname],
+        brokerClusterKey: {
+          name: broker.name,
+          group: broker.group,
+        },
+      };
+      await wkApi.create(worker);
+      await objectApi.create(worker);
+      // worker api will make sure the service is starting, we can skip the checking
+      const wkRes = await wkApi.start(worker);
+      if (!wkRes.errors) result.worker = wkRes.data;
+
+      const workspace = {
+        name: workspaceName,
+        group: workspaceGroup,
+        tags: {
+          name: workspaceName,
+          group: workspaceGroup,
+          nodeNames: [node.hostname],
+        },
+      };
+      objectApi.create(workspace);
+
+      if (withTopic) {
+        const topic = {
+          name: generate.serviceName({ prefix: 'topic' }),
+          group: hashByGroupAndName(workspaceGroup, workspaceName),
+          nodeNames: [node.hostname],
+          brokerClusterKey: {
+            name: broker.name,
+            group: broker.group,
+          },
+          tags: {
+            type: 'shared',
+            parentKey: {
+              name: workspaceName,
+              group: workspaceGroup,
+            },
+          },
+        };
+
+        await topicApi.create(topic);
+        // topic api will make sure the service is starting, we can skip the checking
+        const topicRes = await topicApi.start(topic);
+        if (!topicRes.errors) result.topic = topicRes.data;
+      }
+    }
+
+    return result;
+  },
+);
+
+// Note: this custom command is a "heavy" command, may take almost 40 seconds to accomplish
+// make sure you have set enough timeout of defaultCommandTimeout in cypress.e2e.json
+Cypress.Commands.add(
+  'produceTopicData',
+  // make this command to be child command (we need to create topic first)
+  { preSubject: 'true' },
+  async (workerName, topic) => {
+    const connector = {
+      name: generate.serviceName({ prefix: 'perf' }),
+      // it is ok to use random group here; we just need a temp connector to produce data
+      group: generate.serviceName({ prefix: 'group' }),
+      connector__class: connectorApi.connectorSources.perf,
+      columns: ['name', 'value'],
+      topicKeys: [
+        {
+          name: topic.name,
+          group: topic.group,
+        },
+      ],
+      workerClusterKey: {
+        name: workerName,
+        group: workerGroup,
+      },
+    };
+
+    const connectorRes = await connectorApi.create(connector);
+    if (connectorRes.errors) throw new Error(connectorRes.title);
+
+    await connectorApi.start(connector);
+
+    // sleep 10 seconds for perf connector writes some data to topic
```

Comment:
would rephrase this to:

```suggestion
    // sleep 10 seconds for perf connector to write some data to topic
```

:)